### DISCUSSION TITLE
[8 aprile 2024] Automi

### DISCUSSION THREAD

**Discussion by Antaarees** [sapienzastudentsnetwork/automi-calcolabilita-complessita#25] [Automi]
![photo_2024-04-10_17-16-39](https://github.com/sapienzastudentsnetwork/acc2324/assets/19594279/4768d2f3-25f1-4cfb-ab0a-ec733b473561)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo le seguenti due soluzioni presenti nei miei appunti:

![image](https://github.com/user-attachments/assets/111ef8d3-2805-49e0-9f38-b933b538b48e)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo.

**Reply by FeddyLix17** [reply]
nei 4 sottocasi, invece di $p^k$ non dovrebbe essere $a^k$?

**Reply by Exyss** [reply]
Oops, typo

**Comment by cornflexxx** [comment]
Supponiamo che $L$ sia regolare e consideriamo la stringa $w = c^4a^pb^p\in L$, dove $p$ è la lunghezza del pumping.  
Essendo che $|xy|\le p$ allora $y$ di certo non potrà mai contenere delle $b$, né tantomeno delle $c$ in quanto sono in numero fissato, quindi non possiamo iterarle per un $i$ arbitrario. Quindi $y$ può contenere solo delle $a$, al più $y = a^{p-4}$.

Sia $x = c^4a^{p-4-k}$, $y = a^k$ e $z = a^4b^p$, in questo caso facendo pumping down otterremo $xz = c^4a^{p-k}b^p$, dove $k>0$, e ovviamente $p-k<p$.

**Reply by AglaiaNorza** [reply]
Il fatto che "y non può contenere c" è errato, y può avere delle c - tu stai assumendo x = "cccc" (e quindi che $y = a^{p-4}$), ma non lo puoi sapere a priori. Il pumping lemma deve non-valere per tutte le scomposizioni possibili della stringa che proponi, non puoi decidere tu come scomporla.
In realtà il caso in cui y contiene c è anche "favorevole" perché qualsiasi pumping rompe la regola $c^4$.

**Comment by CiottoloMaggico** [comment]
<img width="1775" height="786" alt="image" src="https://github.com/user-attachments/assets/2598c9d9-b30d-4534-b50c-4aa68ff27ab1" />

**Comment by daromacarlo** [comment]
<img width="1251" height="681" alt="immagine" src="https://github.com/user-attachments/assets/2d0a2dd1-09e0-4eea-9c7b-fd3f4fdb7d47" />
